launch/launch/frontend/expose.py

```diff
@@ -40,13 +40,16 @@ def instantiate_action(entity: 'Entity', parser: 'Parser') -> Action:
 
 def instantiate_substitution(
     type_name: Text,
-    args: Iterable[SomeSubstitutionsType]
+    args: Iterable[Iterable[SomeSubstitutionsType]]
 ) -> Substitution:
     """Call the registered substitution parsing method, according to `args`."""
     if type_name not in substitution_parse_methods:
         raise RuntimeError(
             'Unknown substitution: {}'.format(type_name))
-    subst_type, kwargs = substitution_parse_methods[type_name](*args)
+    if args:
+        subst_type, kwargs = substitution_parse_methods[type_name](*args)
+    else:
+        subst_type, kwargs = substitution_parse_methods[type_name]([])
```

Comment:
@ivanpauno hmm, `substitution_parse_methods[type_name](*args)` should be enough.

Reply:
check my other comment

Reply:
Taking a second look at this, something's odd about the star expansion. Why do we end up with a list that has a single list of normalized substitutions (lists themselves)?

Reply:
Yeap, check https://github.com/ros2/launch/pull/273/commits/816b94a71ff7e2e4eefac94ab0ffc55f79bb2684.